docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,278 @@
+      "size": 731,
+      "annotations": {
+        "org.opencontainers.image.created": "2023-11-22T07:27:41Z"
+      },
+      "artifactType": "application/vnd.oci.empty.v1+json",
+      "reference": "localhost:5000/hello@sha256:1b82e249d83eb4881b8bf4ff9cf13a28799907ddc624b4c3c9140fa77d54fa42"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:28653e2bb5b5a75393c3a8b58ed9998796299b41dc1ff1f55b9f0844ad7ba39c",
+      "size": 630,
+      "annotations": {
+        "org.opencontainers.image.created": "2023-11-25T10:32:54Z"
+      },
+      "artifactType": "application/vnd.oci.empty.v1+json",
+      "reference": "localhost:5000/hello@sha256:28653e2bb5b5a75393c3a8b58ed9998796299b41dc1ff1f55b9f0844ad7ba39c"
+    }
+  ]
+}
+```
+
+### oras manifest fetch
+
+Fetch a specified layer and show the formatted JSON of the fetched manifest in formatted JSON.
+
+```shell
+oras manifest fetch $REGISTRY/$REPO:$TAG --format {{.Layers[0].Reference}}
+```
+
+```json
+{
+    "reference": "$REGISTRY/$REPO:$manifest_digest",
+    "layers" : [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+            "size": 12,
+            "annotations": {
+                "org.opencontainers.image.created":"2023-11-29T06:32:43Z"
+            },
+            "reference": "$REGISTRY/$REPO:$layer0_digest",
+        }
+    ]
+}
+```
+
+### Format the output using the given Go template
+
+In order to custom the manifest fields in the output, format the output using the given [Go template](https://golang.org/pkg/text/template/). The keys of the returned JSON can be used as the values for the `--format` flag.
+
+For example, push an artifact to a repository and show the `reference` and `artifactType` of the pushed artifact in the standard output.
+
+```shell
+$ oras push $REGISTRY/$REPO:$TAG --format "{{.Reference}}', '{{.ArtifactType}}"
+```
+
+```console
+"localhost:5000@sha256:85438e6598bf35057962fff34399a362d469ca30a317939427fca6b7a289e70d, "application/vnd.example+type"  
+```
+
+For example, pull a file and display the specified fields `mediaType`, `reference`, `size` of the pulled file in the standard output.
+
+```shell
+$ oras pull $REGISTRY/$REPO:$TAG --format "{{.MediaType}}, {{.Reference}}, {{.Size}}"
+```
+
+
+```console
+"application/vnd.oci.image.layer.v1.tar","sha256:85438e6598bf35057962fff34399a362d469ca30a317939427fca6b7a289e70d", 12
+```
+
+For example, get the specified annotation value of an artifact by the key name `org.opencontainers.image.created`, with the [index function](https://pkg.go.dev/text/template#pkg-functions) defined in Go template.
+
+```shell
+$ oras discover localhost:5000/hello:v1 --format '{{index .Manifest.Annotations "org.opencontainers.image.created"}}'
+
+```
+
+```console
+"2023-11-29T06:32:43Z"
+```
```

Comment:
Merging the command and result into a console code block would be a good option:
```suggestion
```shell
$ oras discover localhost:5000/hello:v1 --format '{{index .Manifest.Annotations "org.opencontainers.image.created"}}'
"2023-11-29T06:32:43Z"
```
```

Reply:
Same reason as above